src/lib/prismaExtensions.ts

```diff
@@ -0,0 +1,64 @@
+import { Prisma } from '@prisma/client'
+
+/**
+ * Prisma extension for soft delete
+ */
+export const softDelete = Prisma.defineExtension({
+  name: 'softDelete',
+  model: {
+    $allModels: {
+      async delete<M, A>(this: M, args: Prisma.Args<M, 'delete'>['where']): Promise<Prisma.Result<M, A, 'update'>> {
+        const context = Prisma.getExtensionContext(this)
+
+        return (context as any).update({
+          where: { ...args.where },
+          data: {
+            deletedAt: new Date(),
+          },
+        })
+      },
+    },
+  },
+})
+
+/**
+ * Prisma extension for bulk soft delete
+ */
+export const softDeleteMany = Prisma.defineExtension({
+  name: 'softDeleteMany',
+  model: {
+    $allModels: {
+      async deleteMany<M, A>(
+        this: M,
+        args: Prisma.Args<M, 'deleteMany'>['where'],
+      ): Promise<Prisma.Result<M, A, 'updateMany'>> {
+        const context = Prisma.getExtensionContext(this)
+
+        return (context as any).updateMany({
+          where: { ...args.where },
+          data: {
+            deletedAt: new Date(),
+          },
+        })
+      },
+    },
+  },
+})
+
+/**
+ * Prisma extension for filtering out soft deleted values from all relevant operations
+ */
+export const filterSoftDeleted = Prisma.defineExtension({
```

Comment:
does this run automatically in each query? or do we have to call this in queries? @rrojan

Reply:
This runs automatically in all queries @sazanrjb